### PR TITLE
Add ability to get effective response URI

### PR DESCRIPTION
**Pull request by fyrts**
Not sure how I would go about reducing the send method to a single expression. Any suggestions?

### CODE REVIEW

File: tests/ZttpTest.php [hunk 1]
```diff
@@ -487,6 +487,14 @@ function can_use_digest_auth()
         $this->assertTrue($response->isOk());
     }
 
+    /** @test */
+    function can_retrieve_effective_uri()
```
Comment (peter279k) [inline tests/ZttpTest.php:491]: It misses `public` hint for this method.

**Comment by peter279k** [comment]
The PR has been conflicts, and it should merge `master` branch.

File: tests/ZttpTest.php (hunk 1 above)
Comment (fyrts) [reply tests/ZttpTest.php:491]: I'm all for adding the `public` hint, but wouldn't that make it inconsistent with the other tests, considering that they don't have any visibility hints declared either?

**Comment by m1guelpf** [comment]
cc @adamwathan any chance of this getting in?

**Comment by adamwathan** [comment]
Can we change `effectiveURI` to `effectiveUri`? We have `asJson` instead of `asJSON` already so I think that would be more consistent. Happy to merge then 👍

**Comment by adamwathan** [comment]
Tagged a new release, thanks.

**Pull request merged**